website/content/tutorials/2-constraints/_index.md

```diff
@@ -0,0 +1,1069 @@
+        <address>2000 K Street NW Washington, DC 20002</address>
+        <website>https://example.com/isacorp/</website>
+      </vendor>
+      <product-name>AwesomeSAS Model 1 Storage Socket</product-name>
+      <type>sas</type>
+    </ata-socket>
+    <memory>
+      <vendor id="vendor3">
+        <name>Massive Memory Corp</name>
+        <address>3000 K Street NW Washington, DC 20003</address>
+        <website>https://example.com/massive-memory-corp/</website>
+      </vendor>
+      <product-name>Model 3 DDR4-3200 8GB (Module 1)</product-name>
+      <byte-size>8589934592</byte-size>
+    </memory>
+    <memory>
+      <vendor id="vendor3">
+        <name>Massive Memory Corp</name>
+        <address>3000 K Street NW Washington, DC 20003</address>
+        <website>https://example.com/massive-memory-corp/</website>
+      </vendor>
+      <product-name>Model 3 DDR4-3200 8GB (Module 2)</product-name>
+      <byte-size>8589934592</byte-size>
+    </memory>
+  </motherboard>
+</computer>
+{{< /highlight >}}
+{{% /tab %}}
+{{% tab %}}
+{{< highlight yaml "linenos=table,hl_lines=33" >}}
+---
+computer:
+  id: awesomepc1
+  vendor:
+    id: vendor1
+    name: AwesomeComp Incorportated
+    address: 1000 K Street NW Washington, DC 20001
+    website: https://example.com/awesomecomp/
+  motherboard:
+    vendor:
+      id: vendor2
+      name: ISA Corp
+      address: 2000 K Street NW Washington, DC 20002
+      website: https://example.com/isacorp/
+    product-name: Magestic Model M-Ultra Motherboard
+    type: atx
+    cpu:
+      vendor:
+        id: vendor2
+        name: ISA Corp
+        address: 2000 K Street NW Washington, DC 20002
+        website: https://example.com/isacorp/
+      architecture: arm64
+      product-name: Superchip Model 1 4-core Processor
+      speed: 4.7 gigahertz
+    ata-socket:
+      vendor:
+        id: vendor2
+        name: ISA Corp
+        address: 2000 K Street NW Washington, DC 20002
+        website: https://example.com/isacorp/
+      product-name: AwesomeSAS Model 1 Storage Socket
+      type: sas
+    memory-modules:
+    - vendor:
+        id: vendor3
+        name: Massive Memory Corp
+        address: 3000 K Street NW Washington, DC 20003
+        website: https://example.com/massive-memory-corp/
+      product-name: Model 3 DDR4-3200 8GB (Module 1)
+      byte-size: 8589934592
+    - byte-size: 8589934592
+      product-name: Model 3 DDR4-3200 8GB (Module 2)
+      vendor:
+        address: 3000 K Street NW Washington, DC 20003
+        id: vendor3
+        name: Massive Memory Corp
+        website: https://example.com/massive-memory-corp/
+{{< /highlight >}}
+{{% /tab %}}
+{{< /tabs >}}
+
+## Conclusion
+
+In this tutorial, we examined an example of a real-world information model in a domain and how we would model it with a community of stakeholders. We incrementally improved a Metaschema module to add more precision to values for field and flag elements with the use of constraints.
```

Comment:
It might be good to provide a complete example at the end with both allow-value groups included.

Also, it may make sense to more fully indicate what was learned in summary.

Something like:

At this point you know should understand how:
- To express a collection of allowed values
- To use allow-other=no to restrict other values.
- etc.

Reply:
Will do, I think that is worth doing too.

Reply:
Updated in https://github.com/usnistgov/metaschema/pull/508/commits/889758b6fe05bdbb6c5dd410dc00aabe7d95b787.